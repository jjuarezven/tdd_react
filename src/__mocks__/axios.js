const axiosMock = jest.genMockFromModule("axios");

const mockResponse = {
  data: { shops: [{ location: "test location", address: "test address" }] }
};

axiosMock.get.mockImplementation(req);

function req() {
  return new Promise(function (resolve) {
    axiosMock.delayTimer = setTimeout(() => {
      resolve(mockResponse);
    }, 1000);
  });
}

module.exports = axiosMock;
