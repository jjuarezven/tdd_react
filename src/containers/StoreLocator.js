import React, { Component } from "react";
import Button from "./Button";
import Map from "./Map";
import Header from "../components/Header";
import mapChooser from "../mapChooser.js";
import axios from "axios";

class StoreLocator extends Component {
  constructor(props) {
    super(props);

    this.state = { currentMap: "none.png", shops: [] };

    this.chooseMap = this.chooseMap.bind(this);
  }

  async componentDidMount() {
    let response = axios
      .get("http://localhost:3000/data/shops.json")
      .then((reponse) => {
        this.setState({
          shops: response.data.shops
        });
      })
      .catch(() => {
        console.log("error");
      });
  }

  chooseMap(event) {
    this.setState({ currentMap: mapChooser(event.target.value) });
  }

  render() {
    const storeButtons = this.state.shops.map((shop, id) => {
      return (
        <Button
          key={id}
          location={shop.location}
          handleClick={this.chooseMap}
        />
      );
    });

    return (
      <>
        <Header />
        <div>{storeButtons}</div>
        <Map imagename={this.state.currentMap} location={this.props.location} />
      </>
    );
  }
}

export default StoreLocator;
