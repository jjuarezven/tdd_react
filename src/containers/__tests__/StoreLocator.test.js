import React from "react";
import { shallow } from "enzyme";
import StoreLocator from "../StoreLocator";
import axios from "axios";
import renderer from "react-test-renderer";

describe("StoreLocator", () => {
  let mountedStoreLocator;

  beforeEach(() => {
    mountedStoreLocator = shallow(<StoreLocator />);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<StoreLocator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls axios.get in #componentDidMount", () => {
    mountedStoreLocator
      .instance()
      .componentDidMount()
      .then(() => {
        expect(axios.get).toHaveBeenCalled();
      });
  });

  it("calls axios.get with correct url", () => {
    mountedStoreLocator
      .instance()
      .componentDidMount()
      .then(() => {
        expect(axios.get).toHaveBeenCalledWith(
          "http://localhost:3000/data/shops.json"
        );
      });
  });

  it("updates state with api data", () => {
    mountedStoreLocator
      .instance()
      .componentDidMount()
      .then(() => {
        expect(mountedStoreLocator.state()).toHaveProperty("shops", [
          {
            location: "test location",
            address: "test address"
          }
        ]);
      });
  });

  it("renders without crashing", () => {
    expect(mountedStoreLocator.length).toBe(1);
  });

  /* it("renders 3 buttons", () => {
    const buttons = mountedStoreLocator.find("Button");
    expect(buttons.length).toBe(3);
  }); */

  it("renders a map", () => {
    const map = mountedStoreLocator.find("Map");
    expect(map.length).toBe(1);
  });
});

describe("choose map", () => {
  it("updates this.state.currentMap using the location passed to it", () => {
    let mountedStoreLocator = shallow(<StoreLocator />);
    let mockEvent = { target: { value: "testland" } };
    mountedStoreLocator.instance().chooseMap(mockEvent);
    expect(mountedStoreLocator.instance().state.currentMap).toBe(
      "testland.png"
    );
  });
});
