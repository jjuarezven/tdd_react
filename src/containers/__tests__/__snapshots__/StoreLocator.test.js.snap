// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator renders correctly 1`] = `
Array [
  <div
    className="Header"
  >
    <img
      alt="wired brain"
      src="images/wired-brain-coffee-logo.png"
    />
  </div>,
  <div />,
  <div
    className="MapBox"
  >
    <img
      src="images/none.png"
    />
  </div>,
]
`;
